go/parser/parser.go

```diff
@@ -172,7 +172,7 @@ func (p *parser) declare1(decl ast.Node, scope *ast.Scope, kind ast.ObjKind, ide
 				case *ast.StarExpr:
 					rt = t.X.(*ast.Ident).Obj
 				default:
```

Comment:
I think that having the default panic is reasonable. How about having a special
case for *ast.BadExpr instead?

    case *ast.BadExpr:
        return


Then the intent of the code is clearer, I think.